src/os/bluestore/BlueStore.cc

```diff
@@ -6521,16 +6537,30 @@ void BlueStore::_txc_finish_kv(TransContext *txc)
     finishers[n]->queue(txc->onreadable);
     txc->onreadable = NULL;
   }
-  if (txc->oncommit) {
-    finishers[n]->queue(txc->oncommit);
-    txc->oncommit = NULL;
-  }
-  while (!txc->oncommits.empty()) {
-    auto f = txc->oncommits.front();
-    finishers[n]->queue(f);
-    txc->oncommits.pop_front();
+  if (g_conf->bluestore_rtc_sync_completions && txc->kv_submitted_sync &&
+      txc->wal_txn && txc->osr->is_finisher_done()) {
```

Comment:
This Problem is resolved by #11844
The destructors are called only by finishers.